wp-graphql.php

```diff
@@ -287,6 +287,11 @@ private function actions() {
 				$tracker->delete_timestamp();
 			} );
 
+      /**
+       * Set up router actions/filters as early as possible
+       */
+			new \WPGraphQL\Router();
```

Comment:
So, the primary reason this is hooked in to `after_setup_theme` is for themes functions.php to be able to add functions that modify behavior if needed. If the router is setup before this hook, then filters and actions in the Router construct (https://github.com/wp-graphql/wp-graphql/blob/develop/src/Router.php#L44-L82) cannot be hooked into from themes.

Reply:
Could you describe the scenario you have mind? I'm not quite following your reasoning (which means I must be missing something). As I see it, the Router constructor is setting up the endpoint, most likely set through settings, and a few filters and actions. If functions.php is loaded after this is set up, these filters and actions have not run yet so they can still be hooked into.

Separate question: do you think there is an issue with The Events Calendar plugin in that it's forcing user authentication before after_setup_theme has run?

Reply:
This change will instantiate the Router class as soon as WPGraphQL is instantiated, which happens when the plugin is loaded, prior to themes being loaded. 

This means that any code inside a themes functions.php (or other files included by it) would not hav executed yet. 

So, if a theme had a filter such as: 

```
add_filter( 'graphql_endpoint', function() {
    return 'my_custom_endpoint';
});
```

Their code would not work, because the filter (`graphql_endpoint`) would have already been called at WPGraphQL instantiation when the Router is instantiated, and not after themes have been loaded. 

I think there might be a path forward to moving some of the data that's currently in the Router around a bit so we can make both cases work (auth being respected _and_ filters being hookable by themes). 

I'll need to think on this a bit, but the information you've provided and this PR helps get things close for sure!

Reply:
Thanks for taking the time to explain, appreciated and makes sense.

I also submitted a ticket with The Events Calendar plugin support just to let them know of the issue. Not sure if they could easily change their code. Wish WP was a bit more opinionated about what happens when.

Reply:
I keep turning this in my head. I'm thinking that a similar issue would happen if one were to try to do a REST API request with a rest_url_prefix filter in functions.php (which I have). Let me investigate that, will report my findings.

Reply:
Finally was able to pinpoint why it works with the REST API. They do this:

```PHP
  public function serve_request( $path = null ) {
    /* @var WP_User|null $current_user */
    global $current_user;

    if ( $current_user instanceof WP_User && ! $current_user->exists() ) {
      /*
       * If there is no current user authenticated via other means, clear
       * the cached lack of user, so that an authenticate check can set it
       * properly.
       *
       * This is done because for authentications such as Application
       * Passwords, we don't want it to be accepted unless the current HTTP
       * request is a REST API request, which can't always be identified early
       * enough in evaluation.
       */
      $current_user = null;
    }
```

I think this is needed for WP-GraphQL too.

Reply:
New PR in wp-graphql/wp-graphql#1693.

Reply:
@therealgilles ooh! Good find. Shall we close this PR in favor of the other then?

Reply:
Yes. I opted to create a new PR instead of trying to fix up this one, seemed cleaner.